graph/appender/istio_details.go

```diff
@@ -111,3 +114,24 @@ NODES:
 		}
 	}
 }
+
+// addLabels is a chance to add any missing label info to nodes when the telemetry does not provide enough information.
+func addLabels(trafficMap graph.TrafficMap, globalInfo *GlobalInfo, namespaceInfo *NamespaceInfo) {
+	appLabelName := config.Get().IstioLabels.AppLabelName
+	for _, n := range trafficMap {
+		// make sure service nodes have the defined app label so it can be used for app grouping in the UI.
+		if n.NodeType == graph.NodeTypeService && n.App == "" {
+			service, err := globalInfo.Business.Svc.GetServiceDefinition(namespaceInfo.Namespace, n.Service)
+			if err != nil {
+				log.Debugf("Error fetching service definition, may not apply app label correctly for namespace=%s svc=%s: %s", namespaceInfo.Namespace, n.Service, err.Error())
+				if service == nil {
+					continue
+				}
+			}
+
+			if app, ok := service.Service.Labels[appLabelName]; ok {
```

Comment:
Note that Service.Labels[] may be different of labels used for selector.
Just to avoid potential pitfalls as it may be different from the labels used in the pods.
I know this is intentional but I wonder if this distinction is clear enough for the user.